core/src/test/java/org/elasticsearch/index/mapper/RangeFieldTypeTests.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.mapper;
+
+import com.carrotsearch.randomizedtesting.generators.RandomPicks;
+import org.apache.lucene.document.DoubleRangeField;
+import org.apache.lucene.document.FloatRangeField;
+import org.apache.lucene.document.IntRangeField;
+import org.apache.lucene.document.LongRangeField;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.geo.ShapeRelation;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.mapper.RangeFieldMapper.RangeType;
+import org.elasticsearch.index.query.QueryShardContext;
+import org.elasticsearch.test.IndexSettingsModule;
+import org.joda.time.DateTime;
+import org.junit.Before;
+
+public class RangeFieldTypeTests extends FieldTypeTestCase {
+    RangeType type;
+    protected static String FIELDNAME = "field";
+    protected static int DISTANCE = 10;
+
+    @Before
+    public void pickType() {
+        type = RandomPicks.randomFrom(random(), RangeType.values());
+    }
+
+    @Override
+    protected RangeFieldMapper.RangeFieldType createDefaultFieldType() {
+        return new RangeFieldMapper.RangeFieldType(type);
+    }
+
```

Comment:
you should override `setupProperties` to add teh custom properties of this field type such as the date formatter so that they are tested, you can have a look at DateFieldTypeTests for instance.